### DISCUSSION TITLE
I can't make the color picker work in the slate plugin

### DISCUSSION THREAD

**Discussion by ledahu05** [react-page/react-page#1295] [Q&A]
Hi,

I'm trying to add the color plugin to the slate editor:

```
import { ColorPickerField } from '@react-page/editor';
import slate, { pluginFactories } from '@react-page/plugins-slate';
import FormatColorTextIcon from '@mui/icons-material/FormatColorText';
import React from 'react';

const colorPickerPlugin = pluginFactories.createComponentPlugin<{
  color: string;
}>({
  addHoverButton: true, // whether to show it above the text when selected
  addToolbarButton: true, // whether to show it in the bottom toolbar
  type: 'SetColor', // a well defined string, this is kind of the id of the plugin
  object: 'mark', // mark is like a span, other options are inline and block
  icon: <FormatColorTextIcon />, // an icon to show
  label: 'Set Color',
  Component: 'span', // the component to render
  getStyle: ({ color }) => ({ color }),
  controls: {
    // identical to custom cell plugins
    type: 'autoform',
    schema: {
      type: 'object',
      required: ['color'],
      properties: {
        color: {
          uniforms: {
            component: ColorPickerField,
          },
          default: 'rgba(0,0,255,1)',
          type: 'string',
        },
      },
    },
  },
});

export const customSlate: any = slate((config) => ({
  ...config,
  plugins: {
    ...config.plugins,
    color: {
      colorPicker: colorPickerPlugin,
    },
  },
}));

//in my nextjs page
const cellPlugins = [customSlate];
 <Editor
          cellPlugins={cellPlugins}
          value={value}
          onChange={setValue}
          uiTranslator={uiTranslator}
          components={{ BottomToolbar: CustomBottomToolbar }}
/>
```

Everything is rendered properly and all features from the slate cell plugin that doesn't require extra popup works property. 
But the color picker is not working when i click on it (**# plugin and link plugin are ko too**). 
**After debug i noticed that the props onChange received by ColorPickerField is null**. 

I'm obviously missing a step here can you help me please?

The version i'm using are:
`   
 "@react-page/editor": "^5.4.4",
 "@react-page/plugins-slate": "^5.4.4",
`

**Comment by katerynapadafa** [comment]
interested in this issue as well

**Reply by katerynapadafa** [reply]
how i did it 


const colorPickerPlugin=pluginFactories.createComponentPlugin({
  addHoverButton: true,
  addToolbarButton: true,
  type: 'SetColor',
  object: 'mark',
  icon: <Icon type="font-colors" />,
  label: 'Set Color',
  Component: 'span',
  getStyle: ({ color }) => ({ color }),
  controls: {
    type: 'autoform',
    schema: {
      type: 'object',
      required: ['color'],
      properties: {
        color: {
          uniforms: {
            component: ColorPickerField,
          },
          default: 'rgba(0,0,255,1)',
          type: 'string',
        },
      },
    },
  },
})

const customSlate = slate((config) => ({
  ...config,
  plugins: {
    ...config.plugins,
    color: {
      colorPicker: colorPickerPlugin,
    },
  },
}));

const plugins = [ customSlate, image, background(), spacer, divider ]

   <Editor
      cellPlugins={plugins}
      value={theCurrentValue}
      onChange={newValue => saveTheValue(newValue)}
    />
    </div>

**Reply by katerynapadafa** [reply]
image, background(), spacer, divider - are additional plugins

**Reply by ledahu05** [reply]
have you tried to remove the StrictMode ?

**Reply by katerynapadafa** [reply]
my code actually worked

**Comment by IN2TEC** [comment]
Hi there, I am having the same issue. Can someone point me in the right direction?In fact I am receiving an error stating that value cannot be null

![Screenshot 2023-10-05 at 9 56 11 AM](https://github.com/react-page/react-page/assets/104642539/fec943dc-6842-46f4-a30a-480c55467549)

when entering a value I receive a length error and the app crashes

<img width="1112" alt="Screenshot 2023-10-05 at 9 57 11 AM" src="https://github.com/react-page/react-page/assets/104642539/65d44eb5-531e-4aef-bf4e-d374e6832341">

```
  // Plugins for @react-page editor
  const colorPickerPlugin = pluginFactories.createComponentPlugin({
    addHoverButton: true, // whether to show it above the text when selected
    addToolbarButton: true, // whether to show it in the bottom toolbar
    type: 'SetColor', // a well defined string, this is kind of the id of the plugin
    object: 'mark', // mark is like a span, other options are inline and block
    icon: <PaletteIcon />, // an icon to show
    label: 'Set Color',
    value: '',
    Component: 'span', // the component to render
    getStyle: ({ color }) => ({ color }),
    controls: {
      // identical to custom cell plugins
      type: 'autoform',
      schema: {
        type: 'object',
        required: ['color'],
        properties: {
          color: {
            uniforms: {
              component: ColorPickerField,
            },
            default: '#333333',
            type: 'string',
          },
        },
      },
    },
  });
        
  const customSlate = slate((config) => ({
    ...config,
    plugins: {
      ...config.plugins,
      color: {
        colorPicker: colorPickerPlugin,
      },
    },
```
  }));

**Reply by ledahu05** [reply]
Linked to https://github.com/react-page/react-page/issues/1281 maybe ?
Personnaly i stopped using autoform because of this issue.

**Comment by IN2TEC** [comment]
Update: It was React.StrictMode in my local environment - once I pushed it live it works
